### PR TITLE
Update .gitignore

### PR DESCRIPTION
**Pull request by dolanmiu**
Generated .gitignore from:

https://www.gitignore.io/

### CODE REVIEW

**Comment by reda-alaoui** [comment]
Hello @dolanmiu ,

Thank you for the contribution.

However, there are many things in the proposed gitignore that are not relevants to this project, among them:
- yarn
- grunt
- bower
The patch also removes `.idea` and give it no replacement (I use Jetbrains IntelliJ for instance).

It does not make sense for me to add ignores for files coming from unused frameworks / techs.

I will accept the changes, if the gitignore stays focused on:
- npm
- typescript
- any specific IDE files

**Comment by dolanmiu** [comment]
What do you think about adding these:

https://www.gitignore.io/api/intellij

**Comment by reda-alaoui** [comment]
A lot of people consider that some IntelliJ files living inside `.idea` folder can be followed by git.

For my part, I don't want any IDE file to be followed by git.
So for Jetbrains, let's keep `.idea` and append missing parts from https://www.gitignore.io/api/intellij.

File: .gitignore [hunk 1]
```diff
@@ -67,3 +67,75 @@ typings/
 !.vscode/launch.json
 !.vscode/extensions.json
 .history
+
+### Intellij ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and Webstorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# Ignore entire .idea folder
+.idea
```
Comment (dolanmiu) [inline .gitignore]: Here is ignoring the whole folder

**Comment by dolanmiu** [comment]
I will remove yarn, gulp bower when I get home, later tonight

**Comment by reda-alaoui** [comment]
Thanks !

**Pull request merged**